add cloudformation_assemble, yml->json.

tedder opened the pull request:
This is complementary to the cloudformation module itself, it allows a tree of YML files to be kept, then does a last-minute transformation to JSON. I've attempted to document it, and I wrote about it here too:
- http://awsadvent.tumblr.com/post/106058327586/aws-advent-2014-repeatable-infrastructure-with
- https://github.com/tedder/aws-advent-2014-yml-cloudformation/blob/master/article.md (alt link)

I can't figure out how to generate/test the documentation. `make docsite` doesn't work anymore.

Paging @defionscode as requested on #239.

tedder committed:
add cloudformation_assemble, yml->json.

This is complementary to the cloudformation module itself, it allows a tree of YML files to be kept, then does a last-minute transformation to JSON. I've attempted to document it, and I wrote about it here too:

- http://awsadvent.tumblr.com/post/106058327586/aws-advent-2014-repeatable-infrastructure-with
- https://github.com/tedder/aws-advent-2014-yml-cloudformation/blob/master/article.md (alt link)

I can't figure out how to generate/test the documentation. `make docsite` doesn't work anymore.

Will page defionscode as requested on #239.

juliedavila commented on cloud/amazon/cloudformation_assemble.py:
```diff
@@ -0,0 +1,149 @@
+    module = AnsibleModule(
+        argument_spec = dict(
+            directory=dict(default="cfbuild"),
+            destination=dict(default="/tmp/cfbuild.json"),
```
I don't agree with having a default for this parameter or directory. Also, may need to be specified as required?

juliedavila commented:
I'm trying to think of the best way to deal with the generated json file. Placing it in /tmp as you have there might be reasonable. My initial thoughts (just brainstorming):
1. Have you tested this by placing the fragments in a $role/files folder?
2. There doesn't seem to be support for idempotency. Looks like it would report changed every time, though you don't have changed defined at all so it may just return OK every time.
3. There should be exception handling to account for potentially poorly-formatted YAML/JSON. Also, exceptions should be captured and outputted into a `module.fail_json(msg='Exception Message')` type of syntax.

tedder committed:
iterative improvement per feedback

handled code review issue 2 (explicitly declare lack of idempotency) and 3 (use fail_json, catch parsing errors)

tedder commented:
1. I hadn't ever tried that. It doesn't work. Can you point me at code that does the right thing? path_dwim seems to expect the path stuff to already be worked out; I'm a little unsure what to do. I'm on IRC if that's easier.
2. I'll have it return changed=true for now.
3. Yep. Old module that I'm cleaning up, so it wasn't very "modern" in that way. Fixed.
4. I should have really put this in ansible-modules-extras.

tedder committed:
fix improper boolean

tedder commented:
I'm going to maintain this outside of the Ansible castle.
